### ISSUE TITLE
Can't find egs/timit/s5/exp/dnn4_pretrain-dbn_dnn_al

### ISSUE THREAD

**Issue by vyraun** [mravanelli/pytorch-kaldi#29]
Hi,

I followed the instructions and after running run.sh in the timit example:

I only get dnn4_pretrain-dbn_dnn_al_dev in the exp folder, the corresponding test and train files are not there. So, I can't run the dnn alignments step.

The RESULTS file looks like this:

# Hybrid System (Karel's DNN)
---------------------------------Dev Set------------------------------------------
%WER 17.5 | 400 15057 | 84.6 10.5 4.8 2.2 17.5 98.5 | -0.471 | exp/dnn4_pretrain-dbn_dnn/decode_dev/score_6/ctm_39phn.filt.sys
--------------------------------Test Set------------------------------------------
%WER 18.5 | 192 7215 | 84.2 11.0 4.8 2.7 18.5 100.0 | -1.151 | exp/dnn4_pretrain-dbn_dnn/decode_test/score_4/ctm_39phn.filt.sys
# Hybrid System (Karel's DNN), sMBR training
---------------------------------Dev Set------------------------------------------
%WER 17.3 | 400 15057 | 85.5 10.6 4.0 2.7 17.3 98.5 | -0.696 | exp/dnn4_pretrain-dbn_dnn_smbr/decode_dev_it1/score_5/ctm_39phn.filt.sys
%WER 17.3 | 400 15057 | 85.4 10.7 3.9 2.7 17.3 98.5 | -0.380 | exp/dnn4_pretrain-dbn_dnn_smbr/decode_dev_it6/score_7/ctm_39phn.filt.sys
--------------------------------Test Set------------------------------------------
%WER 18.6 | 192 7215 | 84.2 11.1 4.7 2.8 18.6 100.0 | -0.816 | exp/dnn4_pretrain-dbn_dnn_smbr/decode_test_it1/score_5/ctm_39phn.filt.sys
%WER 18.8 | 192 7215 | 84.7 11.4 3.9 3.5 18.8 100.0 | -0.819 | exp/dnn4_pretrain-dbn_dnn_smbr/decode_test_it6/score_5/ctm_39phn.filt.sys
# sMBR not helpful here...

How to get the code working for the timit example in this case? 

Thanks!

**Comment by TParcollet** [comment]
Hi, Is the problem solved ?

**Comment by TParcollet** [comment]
Ok so you removed the exit 0 in run.sh at line 144 right ? Just run local/nnet/run_dnn.sh as commented on the line 144 that you removed :)

**Comment by vyraun** [comment]
@TParcollet Thanks! I had tried with the local/nnet/run_dnn.sh call as well. The program gets stuck at:

SKIPPING TRAINING... (steps/nnet/train.sh)
nnet already trained : exp/dnn4_pretrain-dbn_dnn/final.nnet (nnet/nnet_dbn_dnn_iter13_learnrate1.5625e-05_tr0.8395_cv1.4695_final_)
# Accounting: time=0 threads=1
# Ended (code 0) at Thu Dec 20 09:12:40 EST 2018, elapsed time 0 seconds
steps/nnet/decode.sh --nj 20 --cmd run.pl --acwt 0.2 exp/tri3/graph data-fmllr-tri3/test exp/dnn4_pretrain-dbn_dnn/decode_test
steps/nnet/decode.sh --nj 20 --cmd run.pl --acwt 0.2 exp/tri3/graph data-fmllr-tri3/dev exp/dnn4_pretrain-dbn_dnn/decode_dev

So, calling the alignment step after this gives a fie not found error. Any way to fix this or inspect more?

**Comment by vyraun** [comment]
Hi @TParcollet  The alignemt step now works, I just ran run_dnn.sh after the MFCC feature extraction step as you suggested. However, I am getting an error in the last step now.

python run_exp.py cfg/TIMIT_baselines/TIMIT_MLP_mfcc_basic.cfg 
- Reading config file......OK!
- Chunk creation......OK!

------------------------------ Epoch 000 / 023 ------------------------------
Training TIMIT_tr chunk = 1 / 5
ERROR: training epoch 0, chunk 0 not done! File exp/TIMIT_MLP_basic/exp_files/train_TIMIT_tr_ep000_ck00.info does not exist.
See exp/TIMIT_MLP_basic/log.log 

Some step is still missing it seems.

**Comment by vyraun** [comment]
Another error while running run_dnn.sh is at this step:

steps/nnet/make_denlats.sh: generating denlats from data 'data-fmllr-tri3/train', putting lattices in 'exp/dnn4_pretrain-dbn_dnn_denlats'
bash: line 1:  4974 Segmentation fault      (core dumped) ( latgen-faster-mapped --beam=18.0 --lattice-beam=10.0 --acoustic-scale=0.2 --max-mem=20000000 --max-active=5000 --word-symbol-table=data/lang/words.txt exp/dnn4_pretrain-dbn_dnn/final.mdl exp/dnn4_pretrain-dbn_dnn_denlats/dengraph/HCLG.fst "ark,s,cs:copy-feats scp:data-fmllr-tri3/train/split20/8/feats.scp ark:- | nnet-forward --no-softmax=true --prior-scale=1.0 --feature-transform=exp/dnn4_pretrain-dbn_dnn/final.feature_transform --class-frame-counts=exp/dnn4_pretrain-dbn_dnn/ali_train_pdf.counts --use-gpu=no exp/dnn4_pretrain-dbn_dnn/final.nnet ark:- ark:- |" scp:exp/dnn4_pretrain-dbn_dnn_denlats/lat.store_separately_as_gz.scp ) 2>> exp/dnn4_pretrain-dbn_dnn_denlats/log/decode_den.8.log >> exp/dnn4_pretrain-dbn_dnn_denlats/log/decode_den.8.log
run.pl: 1 / 20 failed, log is in exp/dnn4_pretrain-dbn_dnn_denlats/log/decode_den.*.log

This is why python run_exp.py cfg/TIMIT_baselines/TIMIT_MLP_mfcc_basic.cfg fails.

Anyway to fix the above error?

**Comment by TParcollet** [comment]
Hey, I would need to take a look into ' log is in exp/dnn4_pretrain-dbn_dnn_denlats/log/decode_den.*.log'.

**Comment by TParcollet** [comment]
I would recommend you to do a clean run.sh again. Clean everything generated in your TIMIT folder, call run.sh and then cal run_dnn.sh again. So you don't have multiple run files that could override each others etc ...

**Comment by vyraun** [comment]
> Hey, I would need to take a look into ' log is in exp/dnn4_pretrain-dbn_dnn_denlats/log/decode_den.*.log'.

Hi @TParcollet thanks for the reply. I  have the logs [here](https://www.dropbox.com/sh/us19nb3cz9uomp9/AAC0puXm_CfOQlVHaXaS3YQDa?dl=0)



> I would recommend you to do a clean run.sh again. Clean everything generated in your TIMIT folder, call run.sh and then cal run_dnn.sh again. So you don't have multiple run files that could override each others etc ...

yes, i will try this way.

**Comment by vyraun** [comment]
> The problem is that only 1 job over 20 failed, so you need to find in which .log file there is a 'failed' (At the end of the .lof file)

thanks @TParcollet ! yes, i just checked all log files: none have failed message: 

LOG (latgen-faster-mapped[5.5.151~1420-b126]:main():latgen-faster-mapped.cc:167) Done 184 utterances, failed for 0

> Do you have enough memory to run 20 decoding in parallel ? Are you using queue.pl or cmd ? Could. be a OOM problem, since decoding seems to be working.

I am using run.pl, not queue.pl (replaced them with run.pl in cmd.sh). 

It probably is an oom issue, where can i set the number of parallel decodings?

**Comment by TParcollet** [comment]
In run_dnn.sh you should find some --nj 20, replace them all ;) I don't know what is your actual hardware configuration but do not set it higher than the number of CPU Thread you have. And if you have an high number of CPU thread, be aware of your available RAM. For TIMIT I would consider 2Go of RAM per job (16Go of RAM for --nj 8)
